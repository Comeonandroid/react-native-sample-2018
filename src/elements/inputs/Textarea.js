import React, { PureComponent } from 'react'
import { TextInput } from 'react-native'
import { PropTypes } from 'prop-types'
import styled from 'styled-components'
import { prop, ifProp } from 'styled-tools'

class Textarea extends PureComponent {

  constructor(props) {
    super(props)

    this.handleChange = this.handleChange.bind(this)
    this.handleBlur = this.handleBlur.bind(this)
  }

  handleChange(value) {
    const { onChange, name } = this.props
    if(onChange){
      onChange(name, value)
    }
  }

  handleBlur() {
    const { name, onBlur } = this.props
    if (onBlur) {
      onBlur(name, true)
    }
  }

  render() {
    const { numberOfLines, name, value, error, placeholder, height, ...rest } = this.props
    return(
      <Input
        {...rest}
        name={name}
        value={value}
        placeholder={placeholder}
        onChangeText={this.handleChange}
        onBlur={this.handleBlur}
        multiline
        numberOfLines={numberOfLines}
        error={error}
        height={height}
        placeholderTextColor={'#c1c1c1'}
      />
    )
  }

}

Textarea.propTypes = {
  error: PropTypes.bool,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  numberOfLines: PropTypes.number,
  height: PropTypes.number
}

Textarea.defaultProps = {
  height: 120,
  numberOfLines: 10,
  error: false
}

const Input = styled(TextInput) `
  borderWidth: 1px;
  borderColor: ${ifProp('error', prop('theme.colors.danger'), prop('theme.colors.backgroundMenu'))}
  padding: 8px 10px;
  background: ${ ({ theme }) => theme.colors.backgroundMenu}
  min-height: ${prop('height')};
  color: #fcfcfc;
  font-style: italic;
  font-size: 14px;
`

export default Textarea