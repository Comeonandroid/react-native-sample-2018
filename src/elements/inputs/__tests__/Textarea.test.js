import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'

import { itBehavesLike } from '../../../specs/shared_examples/index'

import Textarea from '../Textarea'

const setup = propOverrides => {
  const props = Object.assign({
    name: 'name',
    value: 'value',
    children: null,
    numberOfLines: 10,
    onChange: jest.fn()
  }, propOverrides)

  const component = <Textarea {...props} />
  const wrapper = shallow(component)

  return {
    props,
    wrapper,
    component
  }
}

describe('<Textarea />', () => {
  const { component } = setup()
  itBehavesLike('aThemedComponent', { subject: component })

  describe('numberOfLines', () => {
    const { component } = setup({ numberOfLines: 15 })
    itBehavesLike('aThemedComponent', { subject: component })
  })
})
