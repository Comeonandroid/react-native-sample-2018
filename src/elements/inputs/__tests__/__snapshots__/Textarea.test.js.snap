// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> dark: renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  error={false}
  height={120}
  multiline={true}
  name="name"
  numberOfLines={10}
  onBlur={[Function]}
  onChange={[MockFunction]}
  onChangeText={[Function]}
  placeholder={undefined}
  placeholderTextColor="#c1c1c1"
  style={
    Array [
      Object {
        "backgroundColor": "#282828",
        "borderColor": "#282828",
        "borderWidth": 1,
        "color": "#fcfcfc",
        "fontSize": 14,
        "fontStyle": "italic",
        "minHeight": 120,
        "paddingBottom": 8,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  value="value"
/>
`;

exports[`<Textarea /> light: renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  error={false}
  height={120}
  multiline={true}
  name="name"
  numberOfLines={10}
  onBlur={[Function]}
  onChange={[MockFunction]}
  onChangeText={[Function]}
  placeholder={undefined}
  placeholderTextColor="#c1c1c1"
  style={
    Array [
      Object {
        "backgroundColor": "#5d5959",
        "borderColor": "#5d5959",
        "borderWidth": 1,
        "color": "#fcfcfc",
        "fontSize": 14,
        "fontStyle": "italic",
        "minHeight": 120,
        "paddingBottom": 8,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  value="value"
/>
`;

exports[`<Textarea /> numberOfLines dark: renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  error={false}
  height={120}
  multiline={true}
  name="name"
  numberOfLines={15}
  onBlur={[Function]}
  onChange={[MockFunction]}
  onChangeText={[Function]}
  placeholder={undefined}
  placeholderTextColor="#c1c1c1"
  style={
    Array [
      Object {
        "backgroundColor": "#282828",
        "borderColor": "#282828",
        "borderWidth": 1,
        "color": "#fcfcfc",
        "fontSize": 14,
        "fontStyle": "italic",
        "minHeight": 120,
        "paddingBottom": 8,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  value="value"
/>
`;

exports[`<Textarea /> numberOfLines light: renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  error={false}
  height={120}
  multiline={true}
  name="name"
  numberOfLines={15}
  onBlur={[Function]}
  onChange={[MockFunction]}
  onChangeText={[Function]}
  placeholder={undefined}
  placeholderTextColor="#c1c1c1"
  style={
    Array [
      Object {
        "backgroundColor": "#5d5959",
        "borderColor": "#5d5959",
        "borderWidth": 1,
        "color": "#fcfcfc",
        "fontSize": 14,
        "fontStyle": "italic",
        "minHeight": 120,
        "paddingBottom": 8,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 8,
      },
      undefined,
    ]
  }
  value="value"
/>
`;
